CRM/CiviAwards/Service/AwardProfile.php

```diff
@@ -28,20 +28,30 @@ public function createProfile($caseTypeId) {
   /**
    * Create fields for a profile.
    *
-   * @param array $customFieldIds
+   * @param array $customFields
    *   Custom fields.
    * @param int $profileId
    *   Profile Id.
    */
-  public function createProfileFields(array $customFieldIds, $profileId) {
-    foreach ($customFieldIds as $customFieldId) {
+  public function createProfileFields(array $customFields, $profileId) {
+    foreach ($customFields as $customField) {
       civicrm_api3('UFField', 'create', [
         'uf_group_id' => $profileId,
-        'field_name' => 'custom_' . $customFieldId,
+        'field_name' => 'custom_' . $customField['id'],
         'field_type' => 'Activity',
-        'label' => 'Activity Field' . $customFieldId,
+        'label' => 'Activity Field' . $customField['id'],
+        'is_required' => boolval($customField['required']),
+        // Nesting api call because of a bug that seems to be if you have two profile fields
+        // in a group and you are about creating another field and you set the weight to say 5.
+        // Civicrm will automatically set it to 3 so that the weight is sequential
+        'api.UFField.create' => [
+          'id' => '$value.id',
+          'weight' => $customField['weight'],
+        ],
       ]);
     }
+    $this->addContactIdField($profileId);
```

Comment:
I don't think we need this `$this->addContactIdField($profileId);` It is already handled outside the loop.